### PR TITLE
Tonga (Assembly): Switch to new scraper

### PR DESCRIPTION
**Pull request by chrismytton**
This updates Tonga Assembly to use a new scraper based on Ruby. It also rescues some members that were going missing or who have changed `id` on the official site.

Scraper change PR: https://github.com/everypolitician-scrapers/tonga-legislative-assembly/pull/1

Fixes https://github.com/everypolitician/everypolitician-data/issues/25303

### Output

```
Add memberships from sources/morph/data.csv
Add memberships from sources/morph/wikidata-members.csv
Merging with sources/archive/term-2015-missing-details.csv
Adding GenderBalance results from sources/gender-balance/results.csv
  ⚥ data for 0; 0 added

Creating names.csv
Persons matched to Wikidata: 0 ✓ | 27 ✘
Parties matched to Wikidata: 0 ✓ | 1 ✘
Areas matched to Wikidata: 0 ✓ | 21 ✘
```

### CODE REVIEW

File: data/Tonga/Assembly/sources/instructions.json [hunk 1]
```diff
@@ -4,13 +4,31 @@
...
+      "file": "morph/wikidata-members.csv",
+      "create": {
+        "from": "morph",
+        "scraper": "everypolitician-scrapers/tonga-wikidata-members",
+        "query": "SELECT item as id, itemlabel as name, itemaltlabel as other_name, date(start) as start_date, date(end) AS end_date, email, 'unknown' AS party, 2015 AS term FROM data"
+      },
+      "type": "membership"
+    },
+    {
+      "file": "archive/data-changed.csv",
```
Comment (tmtmtmtm) [inline data/Tonga/Assembly/sources/instructions.json]: I fear that the name of this file is a little opaque and/or misleading.

File: data/Tonga/Assembly/sources/instructions.json [hunk 2]
```diff
@@ -4,13 +4,31 @@
...
+      "type": "person",
```
Comment (tmtmtmtm) [inline data/Tonga/Assembly/sources/instructions.json]: This source is marked as 'person'-type, but includes 'membership'-type fields: `constituency`, `party`, `term` (https://github.com/everypolitician/everypolitician/issues/589)

File: data/Tonga/Assembly/sources/instructions.json [hunk 3]
```diff
@@ -4,13 +4,31 @@
...
       },
       "source": "http://parliament.gov.to/",
       "type": "membership"
     },
     {
+      "file": "morph/wikidata-members.csv",
+      "create": {
+        "from": "morph",
+        "scraper": "everypolitician-scrapers/tonga-wikidata-members",
+        "query": "SELECT item as id, itemlabel as name, itemaltlabel as other_name, date(start) as start_date, date(end) AS end_date, email, 'unknown' AS party, 2015 AS term FROM data"
```
Comment (tmtmtmtm) [inline data/Tonga/Assembly/sources/instructions.json]: The explicit `'unknown' AS party` here worries me a little. That seems like something that could be tricky to spot later when data for other people, who _do_ have party information is getting clobbered.

File: data/Tonga/Assembly/ep-popolo-v1.0.json [hunk 4]
```diff
@@ -904,17 +995,12 @@
       ]
     },
     {
-      "area_id": "area/vava'u_no._16",
```
Comment (tmtmtmtm) [inline data/Tonga/Assembly/ep-popolo-v1.0.json]: I think you're aware of this anyway, but for completeness, and so we don't overlook it again later, we should be careful not to lose this constituency info.

File: data/Tonga/Assembly/sources/instructions.json (hunk 3 above)
Comment (chrismytton) [inline data/Tonga/Assembly/sources/instructions.json]: Yeah good point, I've removed that for now in https://github.com/everypolitician/everypolitician-data/pull/24902/commits/72c8b3a0fb12f558b6964deea51f6aa0c2ec6e89.

File: data/Tonga/Assembly/sources/instructions.json (hunk 1 above)
Comment (chrismytton) [inline data/Tonga/Assembly/sources/instructions.json]: Hmm, I guess it is. I was going for symmetry with the original `data.csv` from which this is derived, but that's probably just compounding an existing problem :) have renamed it in https://github.com/everypolitician/everypolitician-data/pull/24902/commits/d1ba72120be7cbc6f45117beb7b9bd15b095aca5.

File: data/Tonga/Assembly/sources/instructions.json (hunk 2 above)
Comment (chrismytton) [inline data/Tonga/Assembly/sources/instructions.json]: Removed the `memberships` fields in https://github.com/everypolitician/everypolitician-data/pull/24902/commits/bb0698f2097a7127d8645173031455d0bf1b6497

File: data/Tonga/Assembly/ep-popolo-v1.0.json (hunk 4 above)
Comment (chrismytton) [inline data/Tonga/Assembly/ep-popolo-v1.0.json]: I've rescued this information by putting it into Wikidata, but the `id` has changed slightly, removing the `no._` part. I _think_ that's OK, but thought I'd explicitly flag it up.

**Comment by chrismytton** [comment]
Feedback addressed, so I think this is ready for another review 👍

**Comment by chrismytton** [comment]
This pull request has got a bit long in the tooth. I've opened a new on here - https://github.com/everypolitician/everypolitician-data/pull/28124, so closing this one.

**Pull request closed without merging**